### ISSUE TITLE
[Release 1.11.0] job submission error

### ISSUE THREAD

**Issue by mwtian** [ray-project/ray#22435]
On `releases/1.11.0` branch, there are job submission errors in `rte_ray_client` and `train_small`:
https://buildkite.com/ray-project/periodic-ci/builds/2788#6a73aaf1-80f7-40bf-9b8b-0f21c91e6e57/136-545
https://buildkite.com/ray-project/periodic-ci/builds/2788#1bdebe61-370e-4d93-a979-402732826c34/136-542

These seem like mismatched command handling in product vs in the job client. Can anyone advise on the commit to cherrypick to fix this? e.g. would it be #22011, #22209, or something else? cc @edoakes @simon-mo @krfricke. Assigning to @architkulkarni as Platform oncall.

**Comment by edoakes** [comment]
Oh, I think this is the error that @krfricke fixed awhile back with the command formatting. Let me have a look to see what commit it was.

**Comment by edoakes** [comment]
I believe it's likely this change:
https://github.com/ray-project/ray/commit/dd935874ee3c23ea7636e3853a235ae6c91eabf7

**Comment by architkulkarni** [comment]
> I believe it's likely this change: #22093

Thanks for finding this!  I'll cherry pick this on a test branch off of 1.11.0, run the release test, and see if it fixes the issue.

**Comment by mwtian** [comment]
@krfricke's change has already been cherry-picked into the release branch: https://github.com/ray-project/ray/commit/c500c5b1ed2919de85dd440be1d3550e8e3da9f4. Otherwise almost all tests will fail. But I did not pick https://github.com/ray-project/ray/pull/22011 from `master` which look riskier. Can this be a related issue?

**Comment by mwtian** [comment]
Btw thanks @architkulkarni for testing this.

**Comment by architkulkarni** [comment]
Ah I see, let me test with #22011 instead.

**Comment by simon-mo** [comment]
If it doesn't work, you can bring in this commit which does not ray job submit runners https://github.com/ray-project/ray/commit/ac00389cbe4c58573a6b5949d6eba3bf13387f01

**Comment by mwtian** [comment]
Sounds good. Is this a job issue that needs to be fixed in 1.11.0? Otherwise, https://github.com/ray-project/ray/commit/ac00389cbe4c58573a6b5949d6eba3bf13387f01 / https://github.com/ray-project/ray/pull/22209 looks safer.

**Comment by mwtian** [comment]
I'm ok with either option.

**Comment by architkulkarni** [comment]
#22011 originally fixes a P1 bug, not P0, so by itself it doesn't meet the bar to be cherry-picked.  But if it's the only PR which fixes the release test issue, then we should cherry pick it.

I'll test both https://github.com/ray-project/ray/pull/22011 and https://github.com/ray-project/ray/commit/ac00389cbe4c58573a6b5949d6eba3bf13387f01 / https://github.com/ray-project/ray/pull/22209

**Comment by mwtian** [comment]
Really appreciate it Archit!

**Comment by architkulkarni** [comment]
While I wait for the the wheel build and the test results, just for more context how did you all figure out that it was an issue in the job submission commands?  Naively interpreting the logs it looks like something to do with a missing file: 
```
Command logs:python: can't open file 'wait_cluster.py'
```

And the commands don't have quotes:
```
[INFO 2022-02-15 22:44:36,149] e2e.py: 1736  Running preparation command: python wait_cluster.py 1 600
--
  | [INFO 2022-02-15 22:44:36,150] e2e.py: 1167  Running command in session ses_sPeDTVTnFERNqsA9EUhLtugm:
  | RAY_ADDRESS=auto TEST_OUTPUT_JSON=/tmp/release_test_out.json TEST_STATE_JSON=/tmp/release_test_state.json IS_SMOKE_TEST=0 python wait_cluster.py 1 600
```

**Comment by mwtian** [comment]
IIUC, the previous job command before `wait_cluster.py` call installs `awscli` and copies `wait_cluster.py` and other local files to the Anyscale session: https://github.com/ray-project/ray/blob/8b1bbfe8e438a06bf2f9fe2cbf65f163d64227dd/release/e2e.py#L506-L512. Because the job fails, the `wait_cluster.py` file is missing.

**Comment by architkulkarni** [comment]
Test for #22209 is running here: https://buildkite.com/ray-project/periodic-ci/builds/2803#_ I disabled all tests besides `rte_ray_client`, `train_small` and `many_drivers` (because this PR makes a change to `many_drivers`.)

Test for #22011 is running here: https://buildkite.com/ray-project/periodic-ci/builds/2802# I only ran `rte_ray_client` for this test, but I should have run `train_small` as well.  I can make a separate test for that if we're unsure.

**Comment by architkulkarni** [comment]
@mwtian Based on the tests, both PRs fix the issue, so we can choose which one to use.

The downside of picking #22209 is that we would need to wait for the `many_drivers` tests to finish.  I had to manually fix some merge conflicts here and I'm not sure it's correct.

I would recommend we pick #22011.  The change is very small ([3 lines of code, plus tests](https://github.com/ray-project/ray/pull/22011/files#diff-eecafb0c1cb1630862b364ab4dbb81ec636e3a08107b7fb4a8b32fcb4b13a5c8R166-R168)), and we've validated that it doesn't break the release test infrastructure.  The downside is I don't yet fully understand the root cause, since the S3 upload command doesn't seem to have quotes in it, and Kai's change https://github.com/ray-project/ray/commit/c500c5b1ed2919de85dd440be1d3550e8e3da9f4 already is supposed to address the issue of quotes.
